### DISCUSSION TITLE
Using opacity modifiers with CSS variables assigned to TW theme colors?

### DISCUSSION THREAD

**Discussion by AVGVSTVS96** [tailwindlabs/tailwindcss#12824] [Help] [answered]
I'm working on setting up dynamic theming in my Astro + Tailwind project. I have two sets of colors, primary and accent colors, set in my Tailwind config, both using the default TW color palette but the variables are assigned to the RGB equivalents, not the TW colors themselves.

I was able to implement switching my accent color on the client with CSS variables set to TW color RGB equivalents in my CSS file, then assigning those variables to the accent colors in my TW config:

```cjs
colors: {
  accent: {
	50: 'rgb(var(--accent-50) / <alpha-value>)',
	100: 'rgb(var(--accent-100) / <alpha-value>)',
	200: 'rgb(var(--accent-200) / <alpha-value>)',
	300: 'rgb(var(--accent-300) / <alpha-value>)',
	400: 'rgb(var(--accent-400) / <alpha-value>)',
	500: 'rgb(var(--accent-500) / <alpha-value>)',
	600: 'rgb(var(--accent-600) / <alpha-value>)',
	700: 'rgb(var(--accent-700) / <alpha-value>)',
	800: 'rgb(var(--accent-800) / <alpha-value>)',
	900: 'rgb(var(--accent-900) / <alpha-value>)',
	950: 'rgb(var(--accent-950) / <alpha-value>)',
	}, 
},
```

CSS:
```css
  .sky {
      --accent-50: 240 249 255;
      --accent-100: 224 242 254;
      --accent-200: 186 230 253;
      --accent-300: 125 211 252;
      --accent-400: 56 189 248;
      --accent-500: 14 165 233;
      --accent-600: 2 132 199;
      --accent-700: 3 105 161;
      --accent-800: 7 89 133;
      --accent-900: 12 74 110;
      --accent-950: 8 47 73;
    }
```

Instead of using RGB equivalents for Tailwind colors, I want to use the TW colors themselves. I tried using this syntax:
 `--accent-50: theme(colors.sky[50]`
but this does not work with opacity modifiers, at least not with the syntax I attempted - I removed the rgb() from the TW config but retained the opacity modifier syntax:
`50: 'var(--accent-50) / <alpha-value>'`

I guess this syntax is incorrect when TW colors are set to CSS variables using theme()?

What is the correct way to accomplish this? Is it possible?

**Comment by wongjn** [comment, marked as answer]
The Tailwind colors are defined in hexadecimal format which makes the conventional methods of using CSS variables not available. However, we can leverage [`color-mix()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-mix) instead like:

```
50: 'color-mix(in srgb, var(--accent-50), transparent calc(100% * (1 - <alpha-value>)))',
```
[See this Tailwind Play example](https://play.tailwindcss.com/HgcBj2LUot).

**Reply by AVGVSTVS96** [reply]
Perfect, thanks! I also saw this [example](https://play.tailwindcss.com/svyCppq6HS?file=css) where `const mix` is defined once and used for each shade, so it doesn't have to be repeated.

This is what I ended up with:
```js
const mix = (name) => `color-mix(in srgb, var(${name}), transparent calc(100% * (1 - <alpha-value>)))`;

/** @type {import('tailwindcss').Config} */
export default {
  theme: {
    extend: {
      colors: {
      accent: {
        50: mix('--accent-50'),
        100: mix('--accent-100'),
        200: mix('--accent-200'),
        300: mix('--accent-300'),
        400: mix('--accent-400'),
        500: mix('--accent-500'),
        600: mix('--accent-600'),
        700: mix('--accent-700'),
        800: mix('--accent-800'),
        900: mix('--accent-900'),
        950: mix('--accent-950'),
      }, 
    },
    },
  },
  plugins: [],
}

```

Here's the TW play link for this updated example: https://play.tailwindcss.com/sePx5Lo5et?file=config
